salt/iiif/init.sls

```diff
@@ -0,0 +1,224 @@
+        - require:
+            - loris-config
+            - loris-newrelic-venv
+            - loris-uwsgi-ready
+
+
+# newrelic-python.sls starts running about here
+
+
+loris-uwsgi-config:
+    file.managed:
+        - name: /opt/loris/uwsgi.ini
+        - source: salt://iiif/config/opt-loris-uwsgi.ini
+        - template: jinja
+        - require:
+            - loris-dir
+
+loris-wsgi-entry-point:
+    file.managed:
+        - name: /opt/loris/loris2.wsgi
+        - source: salt://iiif/config/opt-loris-loris2.wsgi
+        - require:
+            - loris-dir
+
+
+
+{% if pillar.elife.env == "dev" %}
+
+# good for development.
+# just clone or move the 'loris-docker' repository into the root of your builder installation.
+build-loris:
+    docker_image.present:
+        - name: elifesciences/loris
+        - tag: latest
+        - build: /vagrant/loris-docker
+        - force: true
+        - require_in:
+            - docker_container: run-loris
+        - onlyif:
+            - test -d /vagrant/loris-docker
+
+{% endif %}        
+
+# `docker_container.running` state:
+# - https://docs.saltstack.com/en/latest/ref/states/all/salt.states.docker_container.html#salt.states.docker_container.running
+run-loris:
+    docker_container.running:
+        - name: loris--{{ pillar.elife.env }} # loris--dev, loris--prod
+        - image: elifesciences/loris
+        - auto_remove: True # "Enable auto-removal of the container on daemon side when the container’s process exits"
+        - hostname: {{ salt['elife.cfg']('project.full_hostname') }} # prod--iiif.elifesciences.org
+        - environment:
+            - NEW_RELIC_ENABLED: {{ pillar.elife.newrelic.enabled }}
+        - port_bindings:
+            - 5004:5004 # uwsgi
+        - binds:
+            # salt-rendered config
+            - /opt/loris/loris2.conf:/opt/loris/etc/loris2.conf
+            - /opt/loris/loris2.wsgi:/var/www/loris2/loris2.wsgi
+            - /opt/loris/uwsgi.ini:/etc/loris2/uwsgi.ini
+            - /opt/loris/newrelic.ini:/etc/newrelic.ini
+            # directories (host:container)
+            # these paths are specified in `loris2.conf`
+            - {{ pillar.iiif.loris.storage }}/tmp:/tmp/loris2/tmp
+            - {{ pillar.iiif.loris.storage }}/cache-resolver:/usr/local/share/images/loris
+            - {{ pillar.iiif.loris.storage }}/cache-general:/var/cache/loris
+            # test directory with a filesystem (fs) resolver
+            - {{ pillar.iiif.loris.storage }}/test:/usr/local/share/images/loris-test
+        - log_driver: syslog # previously /var/log/uwsgi.log
+        - require:
+            - get-loris
+
+            - loris-cache-resolver
+            - loris-cache-general
+            - loris-tmp-directory
+            - loris-cache-blank
+
+            - loris-config
+            - loris-newrelic-venv
+            - loris-uwsgi-config
+            - loris-wsgi-entry-point
+
+            {% if pillar.elife.newrelic.enabled %}
```

Comment:
complex but should work

Reply:
yeah, bit of a wart. I found it was creating a directory called `newrelic.ini` if the newrelic file wasn't generated beforehand (rather than just dying with 'file not found', for example).